### PR TITLE
HL-442: application can be returned from ACCEPTED and REJECTED statuses back to HANDLING

### PR DESCRIPTION
**Pull request by haavikko**
## Description :sparkles:

The application handlers need to be able to revert their decision

## Testing :alembic:
backend/benefit/applications/tests/test_applications_api.py:test_application_status_change_as_handler

### CODE REVIEW

File: backend/benefit/applications/enums.py
Comment (karisal-anders) [inline backend/benefit/applications/enums.py:30]: For uniformity perhaps having the "is not None" handling would be good to have in both if statements or not in either. To me it looks it's not needed at all because cls.values does not contain None and thus "if new status is not None and new_status not in cls.values" is equal to "new_status not in cls.values".

Comment (karisal-anders) [reply backend/benefit/applications/enums.py:30]: Hmm... come to think of Zen of Python:
> Explicit is better than implicit

So having `if new_status is not None` combined with the default value assignment `new_status=None` in the function declaration makes it *explicit that the default value is not accepted*. Because of this argument I accept this code as it is.

File: backend/benefit/applications/tests/test_applications_api.py
Comment (karisal-anders) [inline backend/benefit/applications/tests/test_applications_api.py:900]: How about adding the rest of the cases going back to HANDLING state:

```python
(ApplicationStatus.DRAFT, ApplicationStatus.HANDLING, <status code>),
(ApplicationStatus.RECEIVED, ApplicationStatus.HANDLING, <status code>),
(ApplicationStatus.HANDLING, ApplicationStatus.HANDLING, <status code>),
```

**Review by karisal-anders — APPROVED** [review]
LGTM! If you want you can add a few more test cases but you can merge as is also, I leave that up to you.

File: backend/benefit/applications/enums.py
Comment (haavikko) [reply backend/benefit/applications/enums.py:30]: new_status can be None, in the case that status is not being changed.
status on the other hand can not be None
Added comment to clarify

File: backend/benefit/applications/tests/test_applications_api.py
Comment (haavikko) [reply backend/benefit/applications/tests/test_applications_api.py:900]: The allowed state transitions are defined in the StatusTransitionValidator, I'm only testing a subset of state transitions to ensure the state machine logic works

**Pull request merged**
